src/preferences/PreferencesBase.js

```diff
@@ -189,7 +198,13 @@ define(function (require, exports, module) {
                         try {
                             result.resolve(JSON.parse(text));
                         } catch (e) {
-                            result.reject(new ParsingError("Invalid JSON settings at " + path + "(" + e.toString() + ")"));
+                            if (createIfNew) {
+                                // JSON parsing error -- start from scratch
+                                console.log("Invalid JSON settings at " + path + "(" + e.toString() + ") -- contents reset");
+                                result.resolve({});
```

Comment:
How about creating a backup file "state.json.backup" in the same directory and then deleting all text from the "state.json" file. We can follow it up with a modal dialog box with message- "Contents of corrupted "state.json" file has been cleared. You can recreate the contents manually referring the backup file "state.json.backup" created in the same directory."